saleor/graphql/menu/mutations.py

```diff
@@ -349,18 +350,30 @@ def clean_operation(operation: _MenuMoveOperation):
 
     @classmethod
     def get_operation(
-        cls, info, menu: models.Menu, move: MenuItemMoveInput
+        cls,
+        info,
+        menu_item_to_current_parent,
+        menu: models.Menu,
+        move: MenuItemMoveInput,
     ) -> _MenuMoveOperation:
         menu_item = cls.get_node_or_error(
             info, move.item_id, field="item", only_type="MenuItem", qs=menu.items
         )
         new_parent, parent_changed = None, False
 
+        # we want to check if parent has changes in relation to previous operations
+        # as moves are performed sequentially
+        old_parent_id = (
+            menu_item_to_current_parent[menu_item.pk]
+            if menu_item.pk in menu_item_to_current_parent
+            else menu_item.parent_id
+        )
+
```

Comment:
How about`old_parent_id = menu_item_to_current_parent.get(menu_item.pk, menu_item.parent_id)` ?

Reply:
It cannot be like that, because the parent can be changed to `None` and in such situation the `menu_item.parent_id` will be set as `old_parent_id` instead of `None`.